homeassistant/components/solaredge/config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Config flow for the SolarEdge platform."""
+import solaredge
+import voluptuous as vol
+from requests.exceptions import HTTPError, ConnectTimeout
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.util import slugify
+
+from .const import DOMAIN, DEFAULT_NAME, CONF_SITE_ID
+
+
+@callback
+def solaredge_entries(hass: HomeAssistant):
+    """Return the site_ids for the domain."""
+    return set(
+        (entry.data[CONF_SITE_ID])
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    )
+
+
+class SolarEdgeConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._errors = {}
+
+    def _site_in_configuration_exists(self, site_id) -> bool:
+        """Return True if site_id exists in configuration."""
+        if site_id in solaredge_entries(self.hass):
+            return True
+        return False
+
+    def _check_site(self, site_id, api_key) -> bool:
+        """Check if we can connect to the soleredge api service."""
+        api = solaredge.Solaredge(api_key)
+        try:
+            response = api.get_details(site_id)
+            if response["details"]["status"].lower() != "active":
+                self._errors[CONF_SITE_ID] = "site_not_active"
+                return False
+            return True
+        except KeyError:
+            self._errors[CONF_SITE_ID] = "api_failure"
+            return False
+        except (ConnectTimeout, HTTPError):
```

Comment:
Please move this except up directly after the api call. We should try to wrap as narrowly as possible.